// cm6-theme-gruvbox-dark/dist/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var view = require('@codemirror/view');
var language = require('@codemirror/language');
var highlight = require('@lezer/highlight');

const dark0 = '#282828', dark1 = '#3c3836', dark3 = '#665c54', gray_245 = '#928374', light1 = '#ebdbb2', light2 = '#d5c4a1', light3 = '#bdae93', light4 = '#a89984', bright_red = '#fb4934', bright_green = '#b8bb26', bright_yellow = '#fabd2f', bright_blue = '#83a598', bright_purple = '#d3869b', bright_aqua = '#8ec07c', bright_orange = '#fe8019';
const bg0 = dark0, bg1 = dark1, bg3 = dark3, gray = gray_245, fg1 = light1, fg2 = light2, fg3 = light3, fg4 = light4, red = bright_red, green = bright_green, yellow = bright_yellow, blue = bright_blue, purple = bright_purple, aqua = bright_aqua, orange = bright_orange;
const invalid = red, darkBackground = bg1, highlightBackground = darkBackground, background = bg0, tooltipBackground = bg1, selection = darkBackground, cursor = orange;
/**
The editor theme styles for Gruvbox Dark.
*/
const gruvboxDarkTheme = view.EditorView.theme({
    '&': {
        color: fg1,
        backgroundColor: background
    },
    '.cm-content': {
        caretColor: cursor
    },
    '.cm-cursor, .cm-dropCursor': { borderLeftColor: cursor },
    '&.cm-focused .cm-selectionBackground, .cm-selectionBackground, .cm-content ::selection': { backgroundColor: selection },
    '.cm-panels': { backgroundColor: darkBackground, color: fg1 },
    '.cm-panels.cm-panels-top': { borderBottom: '2px solid black' },
    '.cm-panels.cm-panels-bottom': { borderTop: '2px solid black' },
    '.cm-searchMatch': {
        backgroundColor: bg0,
        color: yellow,
        outline: `1px solid ${bg3}`
    },
    '.cm-searchMatch.cm-searchMatch-selected': {
        backgroundColor: bg3
    },
    '.cm-activeLine': { backgroundColor: highlightBackground },
    '.cm-selectionMatch': { backgroundColor: bg3 },
    '&.cm-focused .cm-matchingBracket, &.cm-focused .cm-nonmatchingBracket': {
        outline: `1px solid ${bg3}`,
        fontStyle: 'bold'
    },
    '&.cm-focused .cm-matchingBracket': {
        backgroundColor: bg3
    },
    '.cm-gutters': {
        backgroundColor: bg1,
        color: fg3,
        border: 'none'
    },
    '.cm-activeLineGutter': {
        backgroundColor: highlightBackground
    },
    '.cm-foldPlaceholder': {
        backgroundColor: 'transparent',
        border: 'none',
        color: '#ddd'
    },
    '.cm-tooltip': {
        border: 'none',
        backgroundColor: tooltipBackground
    },
    '.cm-tooltip .cm-tooltip-arrow:before': {
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent'
    },
    '.cm-tooltip .cm-tooltip-arrow:after': {
        borderTopColor: tooltipBackground,
        borderBottomColor: tooltipBackground
    },
    '.cm-tooltip-autocomplete': {
        '& > ul > li[aria-selected]': {
            backgroundColor: highlightBackground,
            color: fg2
        }
    }
}, { dark: true });
/**
The highlighting style for code in the Gruvbox Dark theme.
*/
const gruvboxDarkHighlightStyle = language.HighlightStyle.define([
    { tag: highlight.tags.keyword, color: red },
    {
        tag: [highlight.tags.name, highlight.tags.deleted, highlight.tags.character, highlight.tags.propertyName, highlight.tags.macroName],
        color: aqua
    },
    { tag: [highlight.tags.variableName], color: blue },
    { tag: [highlight.tags.function(highlight.tags.variableName)], color: green, fontStyle: 'bold' },
    { tag: [highlight.tags.labelName], color: fg1 },
    {
        tag: [highlight.tags.color, highlight.tags.constant(highlight.tags.name), highlight.tags.standard(highlight.tags.name)],
        color: purple
    },
    { tag: [highlight.tags.definition(highlight.tags.name), highlight.tags.separator], color: fg1 },
    { tag: [highlight.tags.brace], color: fg1 },
    {
        tag: [highlight.tags.annotation],
        color: invalid
    },
    {
        tag: [highlight.tags.number, highlight.tags.changed, highlight.tags.annotation, highlight.tags.modifier, highlight.tags.self, highlight.tags.namespace],
        color: purple
    },
    {
        tag: [highlight.tags.typeName, highlight.tags.className],
        color: yellow
    },
    {
        tag: [highlight.tags.operator, highlight.tags.operatorKeyword],
        color: red
    },
    {
        tag: [highlight.tags.tagName],
        color: aqua,
        fontStyle: 'bold'
    },
    {
        tag: [highlight.tags.squareBracket],
        color: orange
    },
    {
        tag: [highlight.tags.angleBracket],
        color: blue
    },
    {
        tag: [highlight.tags.attributeName],
        color: aqua
    },
    {
        tag: [highlight.tags.regexp],
        color: aqua
    },
    {
        tag: [highlight.tags.quote],
        color: gray
    },
    { tag: [highlight.tags.string], color: fg1 },
    {
        tag: highlight.tags.link,
        color: fg4,
        textDecoration: 'underline',
        textUnderlinePosition: 'under'
    },
    {
        tag: [highlight.tags.url, highlight.tags.escape, highlight.tags.special(highlight.tags.string)],
        color: purple
    },
    { tag: [highlight.tags.meta], color: yellow },
    { tag: [highlight.tags.comment], color: gray, fontStyle: 'italic' },
    { tag: highlight.tags.strong, fontWeight: 'bold', color: orange },
    { tag: highlight.tags.emphasis, fontStyle: 'italic', color: green },
    { tag: highlight.tags.strikethrough, textDecoration: 'line-through' },
    { tag: highlight.tags.heading, fontWeight: 'bold', color: green },
    { tag: [highlight.tags.heading1, highlight.tags.heading2], fontWeight: 'bold', color: green },
    {
        tag: [highlight.tags.heading3, highlight.tags.heading4],
        fontWeight: 'bold',
        color: yellow
    },
    {
        tag: [highlight.tags.heading5, highlight.tags.heading6],
        color: yellow
    },
    { tag: [highlight.tags.atom, highlight.tags.bool, highlight.tags.special(highlight.tags.variableName)], color: purple },
    {
        tag: [highlight.tags.processingInstruction, highlight.tags.inserted],
        color: bright_blue
    },
    {
        tag: [highlight.tags.contentSeparator],
        color: red
    },
    { tag: highlight.tags.invalid, color: orange, borderBottom: `1px dotted ${invalid}` }
]);
/**
Extension to enable the Gruvbox Dark theme (both the editor theme and
the highlight style).
*/
const gruvboxDark = [
    gruvboxDarkTheme,
    language.syntaxHighlighting(gruvboxDarkHighlightStyle)
];

exports.gruvboxDark = gruvboxDark;
exports.gruvboxDarkHighlightStyle = gruvboxDarkHighlightStyle;
exports.gruvboxDarkTheme = gruvboxDarkTheme;
