// cm6-theme-gruvbox-light/dist/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var view = require('@codemirror/view');
var language = require('@codemirror/language');
var highlight = require('@lezer/highlight');

const dark1 = '#3c3836', dark2 = '#504945', dark3 = '#665c54', dark4 = '#7c6f64', gray_244 = '#928374', light0 = '#fbf1c7', light1 = '#ebdbb2', light3 = '#bdae93', faded_red = '#9d0006', faded_green = '#79740e', faded_yellow = '#b57614', faded_blue = '#076678', faded_purple = '#8f3f71', faded_aqua = '#427b58', faded_orange = '#af3a03';
const bg0 = light0, bg1 = light1, bg3 = light3, gray = gray_244, fg1 = dark1, fg2 = dark2, fg3 = dark3, fg4 = dark4, red = faded_red, green = faded_green, yellow = faded_yellow, blue = faded_blue, purple = faded_purple, aqua = faded_aqua, orange = faded_orange;
const invalid = red, darkBackground = bg1, highlightBackground = darkBackground, background = bg0, tooltipBackground = bg1, selection = darkBackground, cursor = orange;
/**
The editor theme styles for Gruvbox Light.
*/
const gruvboxLightTheme = view.EditorView.theme({
    '&': {
        color: fg1,
        backgroundColor: background
    },
    '.cm-content': {
        caretColor: cursor
    },
    '.cm-cursor, .cm-dropCursor': { borderLeftColor: cursor },
    '&.cm-focused .cm-selectionBackground, .cm-selectionBackground, .cm-content ::selection': { backgroundColor: selection },
    '.cm-panels': { backgroundColor: darkBackground, color: fg1 },
    '.cm-panels.cm-panels-top': { borderBottom: '2px solid black' },
    '.cm-panels.cm-panels-bottom': { borderTop: '2px solid black' },
    '.cm-searchMatch': {
        backgroundColor: bg0,
        color: yellow,
        outline: `1px solid ${bg3}`
    },
    '.cm-searchMatch.cm-searchMatch-selected': {
        backgroundColor: bg3
    },
    '.cm-activeLine': { backgroundColor: highlightBackground },
    '.cm-selectionMatch': { backgroundColor: bg3 },
    '&.cm-focused .cm-matchingBracket, &.cm-focused .cm-nonmatchingBracket': {
        outline: `1px solid ${bg3}`,
        fontStyle: 'bold'
    },
    '&.cm-focused .cm-matchingBracket': {
        backgroundColor: bg3
    },
    '.cm-gutters': {
        backgroundColor: bg1,
        color: fg3,
        border: 'none'
    },
    '.cm-activeLineGutter': {
        backgroundColor: highlightBackground
    },
    '.cm-foldPlaceholder': {
        backgroundColor: 'transparent',
        border: 'none',
        color: '#ddd'
    },
    '.cm-tooltip': {
        border: 'none',
        backgroundColor: tooltipBackground
    },
    '.cm-tooltip .cm-tooltip-arrow:before': {
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent'
    },
    '.cm-tooltip .cm-tooltip-arrow:after': {
        borderTopColor: tooltipBackground,
        borderBottomColor: tooltipBackground
    },
    '.cm-tooltip-autocomplete': {
        '& > ul > li[aria-selected]': {
            backgroundColor: highlightBackground,
            color: fg2
        }
    }
}, { dark: false });
/**
The highlighting style for code in the Gruvbox Light theme.
*/
const gruvboxLightHighlightStyle = language.HighlightStyle.define([
    { tag: highlight.tags.keyword, color: red },
    {
        tag: [highlight.tags.name, highlight.tags.deleted, highlight.tags.character, highlight.tags.propertyName, highlight.tags.macroName],
        color: aqua
    },
    { tag: [highlight.tags.variableName], color: blue },
    { tag: [highlight.tags.function(highlight.tags.variableName)], color: green, fontStyle: 'bold' },
    { tag: [highlight.tags.labelName], color: fg1 },
    {
        tag: [highlight.tags.color, highlight.tags.constant(highlight.tags.name), highlight.tags.standard(highlight.tags.name)],
        color: purple
    },
    { tag: [highlight.tags.definition(highlight.tags.name), highlight.tags.separator], color: fg1 },
    { tag: [highlight.tags.brace], color: fg1 },
    {
        tag: [highlight.tags.annotation],
        color: invalid
    },
    {
        tag: [highlight.tags.number, highlight.tags.changed, highlight.tags.annotation, highlight.tags.modifier, highlight.tags.self, highlight.tags.namespace],
        color: purple
    },
    {
        tag: [highlight.tags.typeName, highlight.tags.className],
        color: yellow
    },
    {
        tag: [highlight.tags.operator, highlight.tags.operatorKeyword],
        color: red
    },
    {
        tag: [highlight.tags.tagName],
        color: aqua,
        fontStyle: 'bold'
    },
    {
        tag: [highlight.tags.squareBracket],
        color: orange
    },
    {
        tag: [highlight.tags.angleBracket],
        color: blue
    },
    {
        tag: [highlight.tags.attributeName],
        color: aqua
    },
    {
        tag: [highlight.tags.regexp],
        color: aqua
    },
    {
        tag: [highlight.tags.quote],
        color: gray
    },
    { tag: [highlight.tags.string], color: fg1 },
    {
        tag: highlight.tags.link,
        color: fg4,
        textDecoration: 'underline',
        textUnderlinePosition: 'under'
    },
    {
        tag: [highlight.tags.url, highlight.tags.escape, highlight.tags.special(highlight.tags.string)],
        color: purple
    },
    { tag: [highlight.tags.meta], color: yellow },
    { tag: [highlight.tags.comment], color: gray, fontStyle: 'italic' },
    { tag: highlight.tags.strong, fontWeight: 'bold', color: orange },
    { tag: highlight.tags.emphasis, fontStyle: 'italic', color: green },
    { tag: highlight.tags.strikethrough, textDecoration: 'line-through' },
    { tag: highlight.tags.heading, fontWeight: 'bold', color: green },
    { tag: [highlight.tags.heading1, highlight.tags.heading2], fontWeight: 'bold', color: green },
    {
        tag: [highlight.tags.heading3, highlight.tags.heading4],
        fontWeight: 'bold',
        color: yellow
    },
    {
        tag: [highlight.tags.heading5, highlight.tags.heading6],
        color: yellow
    },
    { tag: [highlight.tags.atom, highlight.tags.bool, highlight.tags.special(highlight.tags.variableName)], color: purple },
    {
        tag: [highlight.tags.processingInstruction, highlight.tags.inserted],
        color: blue
    },
    {
        tag: [highlight.tags.contentSeparator],
        color: red
    },
    { tag: highlight.tags.invalid, color: orange, borderBottom: `1px dotted ${invalid}` }
]);
/**
Extension to enable the Gruvbox Light theme (both the editor theme and
the highlight style).
*/
const gruvboxLight = [
    gruvboxLightTheme,
    language.syntaxHighlighting(gruvboxLightHighlightStyle)
];

exports.gruvboxLight = gruvboxLight;
exports.gruvboxLightHighlightStyle = gruvboxLightHighlightStyle;
exports.gruvboxLightTheme = gruvboxLightTheme;
