// cm6-theme-gruvbox-light/dist/index.js
import { EditorView } from '@codemirror/view';
import { HighlightStyle, syntaxHighlighting } from '@codemirror/language';
import { tags } from '@lezer/highlight';

const dark1 = '#3c3836', dark2 = '#504945', dark3 = '#665c54', dark4 = '#7c6f64', gray_244 = '#928374', light0 = '#fbf1c7', light1 = '#ebdbb2', light3 = '#bdae93', faded_red = '#9d0006', faded_green = '#79740e', faded_yellow = '#b57614', faded_blue = '#076678', faded_purple = '#8f3f71', faded_aqua = '#427b58', faded_orange = '#af3a03';
const bg0 = light0, bg1 = light1, bg3 = light3, gray = gray_244, fg1 = dark1, fg2 = dark2, fg3 = dark3, fg4 = dark4, red = faded_red, green = faded_green, yellow = faded_yellow, blue = faded_blue, purple = faded_purple, aqua = faded_aqua, orange = faded_orange;
const invalid = red, darkBackground = bg1, highlightBackground = darkBackground, background = bg0, tooltipBackground = bg1, selection = darkBackground, cursor = orange;
/**
The editor theme styles for Gruvbox Light.
*/
const gruvboxLightTheme = /*@__PURE__*/EditorView.theme({
    '&': {
        color: fg1,
        backgroundColor: background
    },
    '.cm-content': {
        caretColor: cursor
    },
    '.cm-cursor, .cm-dropCursor': { borderLeftColor: cursor },
    '&.cm-focused .cm-selectionBackground, .cm-selectionBackground, .cm-content ::selection': { backgroundColor: selection },
    '.cm-panels': { backgroundColor: darkBackground, color: fg1 },
    '.cm-panels.cm-panels-top': { borderBottom: '2px solid black' },
    '.cm-panels.cm-panels-bottom': { borderTop: '2px solid black' },
    '.cm-searchMatch': {
        backgroundColor: bg0,
        color: yellow,
        outline: `1px solid ${bg3}`
    },
    '.cm-searchMatch.cm-searchMatch-selected': {
        backgroundColor: bg3
    },
    '.cm-activeLine': { backgroundColor: highlightBackground },
    '.cm-selectionMatch': { backgroundColor: bg3 },
    '&.cm-focused .cm-matchingBracket, &.cm-focused .cm-nonmatchingBracket': {
        outline: `1px solid ${bg3}`,
        fontStyle: 'bold'
    },
    '&.cm-focused .cm-matchingBracket': {
        backgroundColor: bg3
    },
    '.cm-gutters': {
        backgroundColor: bg1,
        color: fg3,
        border: 'none'
    },
    '.cm-activeLineGutter': {
        backgroundColor: highlightBackground
    },
    '.cm-foldPlaceholder': {
        backgroundColor: 'transparent',
        border: 'none',
        color: '#ddd'
    },
    '.cm-tooltip': {
        border: 'none',
        backgroundColor: tooltipBackground
    },
    '.cm-tooltip .cm-tooltip-arrow:before': {
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent'
    },
    '.cm-tooltip .cm-tooltip-arrow:after': {
        borderTopColor: tooltipBackground,
        borderBottomColor: tooltipBackground
    },
    '.cm-tooltip-autocomplete': {
        '& > ul > li[aria-selected]': {
            backgroundColor: highlightBackground,
            color: fg2
        }
    }
}, { dark: false });
/**
The highlighting style for code in the Gruvbox Light theme.
*/
const gruvboxLightHighlightStyle = /*@__PURE__*/HighlightStyle.define([
    { tag: tags.keyword, color: red },
    {
        tag: [tags.name, tags.deleted, tags.character, tags.propertyName, tags.macroName],
        color: aqua
    },
    { tag: [tags.variableName], color: blue },
    { tag: [/*@__PURE__*/tags.function(tags.variableName)], color: green, fontStyle: 'bold' },
    { tag: [tags.labelName], color: fg1 },
    {
        tag: [tags.color, /*@__PURE__*/tags.constant(tags.name), /*@__PURE__*/tags.standard(tags.name)],
        color: purple
    },
    { tag: [/*@__PURE__*/tags.definition(tags.name), tags.separator], color: fg1 },
    { tag: [tags.brace], color: fg1 },
    {
        tag: [tags.annotation],
        color: invalid
    },
    {
        tag: [tags.number, tags.changed, tags.annotation, tags.modifier, tags.self, tags.namespace],
        color: purple
    },
    {
        tag: [tags.typeName, tags.className],
        color: yellow
    },
    {
        tag: [tags.operator, tags.operatorKeyword],
        color: red
    },
    {
        tag: [tags.tagName],
        color: aqua,
        fontStyle: 'bold'
    },
    {
        tag: [tags.squareBracket],
        color: orange
    },
    {
        tag: [tags.angleBracket],
        color: blue
    },
    {
        tag: [tags.attributeName],
        color: aqua
    },
    {
        tag: [tags.regexp],
        color: aqua
    },
    {
        tag: [tags.quote],
        color: gray
    },
    { tag: [tags.string], color: fg1 },
    {
        tag: tags.link,
        color: fg4,
        textDecoration: 'underline',
        textUnderlinePosition: 'under'
    },
    {
        tag: [tags.url, tags.escape, /*@__PURE__*/tags.special(tags.string)],
        color: purple
    },
    { tag: [tags.meta], color: yellow },
    { tag: [tags.comment], color: gray, fontStyle: 'italic' },
    { tag: tags.strong, fontWeight: 'bold', color: orange },
    { tag: tags.emphasis, fontStyle: 'italic', color: green },
    { tag: tags.strikethrough, textDecoration: 'line-through' },
    { tag: tags.heading, fontWeight: 'bold', color: green },
    { tag: [tags.heading1, tags.heading2], fontWeight: 'bold', color: green },
    {
        tag: [tags.heading3, tags.heading4],
        fontWeight: 'bold',
        color: yellow
    },
    {
        tag: [tags.heading5, tags.heading6],
        color: yellow
    },
    { tag: [tags.atom, tags.bool, /*@__PURE__*/tags.special(tags.variableName)], color: purple },
    {
        tag: [tags.processingInstruction, tags.inserted],
        color: blue
    },
    {
        tag: [tags.contentSeparator],
        color: red
    },
    { tag: tags.invalid, color: orange, borderBottom: `1px dotted ${invalid}` }
]);
/**
Extension to enable the Gruvbox Light theme (both the editor theme and
the highlight style).
*/
const gruvboxLight = [
    gruvboxLightTheme,
    /*@__PURE__*/syntaxHighlighting(gruvboxLightHighlightStyle)
];

export { gruvboxLight, gruvboxLightHighlightStyle, gruvboxLightTheme };
